### PR TITLE
[GITHUB-2] Adds users and security from code 

### PR DESCRIPTION
**Pull request by lobsterdore**
Adds the ability to configure Nexus via Groovy scripts and adds
more control over the OS service.

Changes:

* Adds tasks for executing Groovy scripts from code
* Adds flag for turning off service start as well as enablement,
this allows for child roles to install plugins configure the service
* Adds Groovy for updating the default admin password
* Adds Groovy for setting up an S3 blobstore
* Adds Groovy for setting up realms and roles
* Adds Groovy for setting HTTP proxies
* Fixes tests and they weren't really testing anything

### CODE REVIEW

File: molecule/default/tests/test_default.py [hunk 1]
```diff
@@ -1,17 +1,36 @@
...
-    # Assert /etc/hosts exists,...
-    f = host.file('/etc/hosts')
-    assert f.exists
-    # ...is owned by the user root,...
-    assert f.user == 'root'
-    # ...and owned by the group root.
-    assert f.group == 'root'
+def test_config(host):
+    nexus_config = host.file(
+        '/home/nexus//sonatype-work/nexus3/etc/nexus.properties'
```
Comment (nathanwebsterdotme) [inline molecule/default/tests/test_default.py]: double slash intentional?

**Review by dennisconrad — APPROVED** [review]

**Review by nathanwebsterdotme — APPROVED** [review]

File: molecule/default/tests/test_default.py (hunk 1 above)
Comment (lobsterdore) [reply molecule/default/tests/test_default.py]: D'oh that's a type, fixing.

File: files/groovy/system_http_proxy.groovy [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import groovy.json.JsonSlurper
+
+parsed_args = new JsonSlurper().parseText(args)
+
+
```
Comment (wojtek-oledzki) [inline files/groovy/system_http_proxy.groovy]: double empty line :(

**Review by wojtek-oledzki — COMMENTED** [review]
and a generic question - can we test "dummy-role"?

File: files/groovy/system_http_proxy.groovy (hunk 2 above)
Comment (lobsterdore) [reply files/groovy/system_http_proxy.groovy]: Fixed

**Comment by wojtek-oledzki** [comment]
Pure Awesomeness!

**Review by wojtek-oledzki — APPROVED** [review]

**Pull request merged**